libs/positions/src/lib/positions-table.spec.tsx

```diff
@@ -1,103 +1,196 @@
-            symbol: 'tDAI',
-            name: 'tDAI TEST',
-            decimals: 5,
-            __typename: 'Asset',
-          },
-          quoteName: 'DAI',
-          __typename: 'Future',
-        },
-        __typename: 'Instrument',
-      },
-      __typename: 'TradableInstrument',
-    },
-    __typename: 'Market',
-  },
-  __typename: 'Position',
+const singleRow: Position = {
+  marketName: 'ETH/BTC (31 july 2022)',
+  averageEntryPrice: '133', // 13.3
+  capitalUtilisation: 11, // 11.00%
+  currentLeverage: 1.1,
+  marketDecimalPlaces: 1,
+  positionDecimalPlaces: 0,
+  assetDecimals: 2,
+  // generalAccountBalance: '0',
+  totalBalance: '123456',
+  assetSymbol: 'BTC',
+  // leverageInitial: '0',
+  // leverageMaintenance: '0',
+  // leverageRelease: '0',
+  // leverageSearch: '0',
+  liquidationPrice: '83', // 8.3
+  // marginAccountBalance: '0',
+  lowMarginLevel: false,
+  // marginMaintenance: '0',
+  // marginSearch: '0',
+  // marginInitial: '0',
+  marketId: 'string',
+  marketTradingMode: MarketTradingMode.Continuous,
+  markPrice: '123', // 12.3
+  notional: '12300', // 1230.0
+  openVolume: '100', // 100
+  realisedPNL: '123', // 1.23
+  unrealisedPNL: '456', // 4.56
+  searchPrice: '0',
+  updatedAt: '2022-07-27T15:02:58.400Z',
 };
+
 const singleRowData = [singleRow];
 
-describe('PositionsTable', () => {
-  it('should render successfully', async () => {
-    act(async () => {
-      const { baseElement } = render(<PositionsTable data={[]} />);
-      expect(baseElement).toBeTruthy();
+it('should render successfully', async () => {
+  act(async () => {
+    const { baseElement } = render(<PositionsTable rowData={[]} />);
+    expect(baseElement).toBeTruthy();
+  });
+});
+
+it('Render correct columns', async () => {
+  act(async () => {
+    render(<PositionsTable rowData={singleRowData} />);
+    await waitFor(async () => {
+      const headers = screen.getAllByRole('columnheader');
+      expect(headers).toHaveLength(9);
+      expect(
+        headers.map((h) =>
+          h.querySelector('[ref="eText"]')?.textContent?.trim()
+        )
+      ).toEqual([
+        'Market',
+        'Amount',
+        'Mark Price',
+        'Entry Price',
+        'Leverage',
+        'Margin allocated',
+        'Realised PNL',
+        'Unrealised PNL',
+        'Updated',
+      ]);
     });
   });
+});
 
-  it('should render correct columns', async () => {
-    act(async () => {
-      render(<PositionsTable data={singleRowData} />);
-      await waitFor(async () => {
-        const headers = screen.getAllByRole('columnheader');
-        expect(headers).toHaveLength(5);
-        expect(headers.map((h) => h.textContent?.trim())).toEqual([
-          'Market',
-          'Amount',
-          'Average Entry Price',
-          'Mark Price',
-          'Realised PNL',
-        ]);
-      });
+it('Splits market name', async () => {
+  act(async () => {
```

Comment:
question (blocking): Not sure `act` usage is correct in these tests? You should only have to wrap the code that causes the update, in this case render (due to ag grid)

```
await act(async () => {
  render(...)
})
expect(screen.getByText('ETH/BTC')).toBeTruthy();
expect(screen.getByText('31 july 2022')).toBeTruthy();
```

Reply:
_I think_ that this should also work and is a bit cleaner with less pyramid of doom.
```
render(...)
expect(await screen.findByTest('ETH/BTC')).toBeTruthy()
```